File: java/src/main/java/com/experian/dvs/client/address/format/AddressEnrichment.java
package com.experian.dvs.client.address.format;

import com.experian.dvs.client.address.format.enrichment.*;
import com.experian.dvs.client.server.address.format.RestApiAddressFormatEnrichment;
import com.experian.dvs.client.server.address.format.RestApiEnrichmentMetadata;
import com.experian.dvs.client.server.address.format.RestApiEnrichmentResultAddress;

import java.util.Objects;
import java.util.Optional;

public class AddressEnrichment {

    private final String transactionId;
    private final Geocodes geocodes;
    private final UsaRegionalGeocodes usaRegionalGeocodes;
    private final AusRegionalGeocodes ausRegionalGeocodes;
    private final NzlRegionalGeocodes nzlRegionalGeocodes;
    private final UKLocationComplete ukLocationComplete;
    private final UKLocationEssential ukLocationEssential;
    private final GbrGovernment gbrGovernment;
    private final GbrBusiness gbrBusiness;
    private final GbrHealth gbrHealth;
    private final Metadata metadata;

    public AddressEnrichment(RestApiAddressFormatEnrichment apiAddressFormatEnrichment) {
        this.transactionId = Objects.toString(apiAddressFormatEnrichment.getTransactionId(), "");
        final RestApiEnrichmentResultAddress result = apiAddressFormatEnrichment.getResult();
        if (result != null) {
            this.geocodes = result.getGeocodes() != null ? new Geocodes(result.getGeocodes()) : null;
            this.usaRegionalGeocodes = result.getUsaRegionalGeocodes() != null ? new UsaRegionalGeocodes(result.getUsaRegionalGeocodes()) : null;
            this.ausRegionalGeocodes = result.getAusRegionalGeocodes() != null ? new AusRegionalGeocodes(result.getAusRegionalGeocodes()) : null;
            this.nzlRegionalGeocodes = result.getNzlRegionalGeocodes() != null ? new NzlRegionalGeocodes(result.getNzlRegionalGeocodes()) : null;
            this.ukLocationComplete = result.getUkLocationComplete() != null ? new UKLocationComplete(result.getUkLocationComplete()) : null;
            this.ukLocationEssential = result.getUkLocationEssential() != null ? new UKLocationEssential(result.getUkLocationEssential()) : null;
            this.gbrGovernment = result.getGbrGovernment() != null ? new GbrGovernment(result.getGbrGovernment()) : null;
            this.gbrBusiness = result.getGbrBusiness() != null ? new GbrBusiness(result.getGbrBusiness()) : null;
            this.gbrHealth = result.getGbrHealth() != null ? new GbrHealth(result.getGbrHealth()) : null;
        } else {
            this.geocodes = null;
            this.usaRegionalGeocodes = null;
            this.ausRegionalGeocodes = null;
            this.nzlRegionalGeocodes = null;
            this.ukLocationComplete = null;
            this.ukLocationEssential = null;
            this.gbrGovernment = null;
            this.gbrBusiness = null;
            this.gbrHealth = null;
        }
        final RestApiEnrichmentMetadata apiEnrichmentMetadata = apiAddressFormatEnrichment.getMetadata();
        if (apiEnrichmentMetadata != null) {
            this.metadata = new Metadata(apiEnrichmentMetadata);
        } else {
            this.metadata = null;
        }
    }

    public String getTransactionId() {
        return transactionId;
    }

    public Optional<Geocodes> getGeocodes() {
        return Optional.ofNullable(geocodes);
    }

    public Optional<UsaRegionalGeocodes> getUsaRegionalGeocodes() {
        return Optional.ofNullable(usaRegionalGeocodes);
    }

    public Optional<AusRegionalGeocodes> getAusRegionalGeocodes() {
        return Optional.ofNullable(ausRegionalGeocodes);
    }

    public Optional<NzlRegionalGeocodes> getNzlRegionalGeocodes() {
        return Optional.ofNullable(nzlRegionalGeocodes);
    }

    public Optional<UKLocationComplete> getUkLocationComplete() {
        return Optional.ofNullable(ukLocationComplete);
    }

    public Optional<UKLocationEssential> getUkLocationEssential() {
        return Optional.ofNullable(ukLocationEssential);
    }

    public Optional<GbrGovernment> getGbrGovernment() {
        return Optional.ofNullable(gbrGovernment);
    }

    public Optional<GbrBusiness> getGbrBusiness() {
        return Optional.ofNullable(gbrBusiness);
    }

    public Optional<GbrHealth> getGbrHealth() {
        return Optional.ofNullable(gbrHealth);
    }
}

File: csharp/DVSClient/DVSClient/address/format/AddressEnrichment.cs
using DVSClient.Address.Format.Enrichment;
using DVSClient.Server.Address.Format;

namespace DVSClient.Address.Format
{
    public class AddressEnrichment
    {
        public string? TransactionId { get; }
        public Geocodes? Geocodes { get; }
        public UsaRegionalGeocodes? UsaRegionalGeocodes { get; }
        public AusRegionalGeocodes? AusRegionalGeocodes { get; }
        public NzlRegionalGeocodes? NzlRegionalGeocodes { get; }
        public UKLocationComplete? GbrLocationComplete { get; }
        public UKLocationEssential? GbrLocationEssential { get; }
        public GbrGovernment? GbrGovernment { get; }
        public GbrBusiness? GbrBusiness { get; }
        public GbrHealth? GbrHealth { get; }
        public Metadata? Metadata { get; }

        public AddressEnrichment(RestApiAddressFormatEnrichment apiAddressFormatEnrichment)
        {
            TransactionId = apiAddressFormatEnrichment.TransactionId ?? string.Empty;

            var result = apiAddressFormatEnrichment.Result;
            Geocodes = result?.Geocodes != null ? new Geocodes(result.Geocodes) : null;
            UsaRegionalGeocodes = result?.UsaRegionalGeocodes != null ? new UsaRegionalGeocodes(result.UsaRegionalGeocodes) : null;
            AusRegionalGeocodes = result?.AusRegionalGeocodes != null ? new AusRegionalGeocodes(result.AusRegionalGeocodes) : null;
            NzlRegionalGeocodes = result?.NzlRegionalGeocodes != null ? new NzlRegionalGeocodes(result.NzlRegionalGeocodes) : null;
            GbrLocationComplete = result?.GbrLocationComplete != null ? new UKLocationComplete(result.GbrLocationComplete) : null;
            GbrLocationEssential = result?.GbrLocationEssential != null ? new UKLocationEssential(result.GbrLocationEssential) : null;
            GbrGovernment = result?.GbrGovernment != null ? new GbrGovernment(result.GbrGovernment) : null;
            GbrBusiness = result?.GbrBusiness != null ? new GbrBusiness(result.GbrBusiness) : null;
            GbrHealth = result?.GbrHealth != null ? new GbrHealth(result.GbrHealth) : null;

            var apiEnrichmentMetadata = apiAddressFormatEnrichment.Metadata;
            Metadata = apiEnrichmentMetadata != null ? new Metadata(apiEnrichmentMetadata) : null;
        }
    }
}